From akpm@linux-foundation.org Mon Feb  4 23:35:34 2008
From: Christoph Lameter <clameter@sgi.com>
Date: Mon, 04 Feb 2008 23:35:48 -0800
Subject: slabinfo: fall back from /sys/kernel/slab to /sys/slab
To: greg@kroah.com
Cc: akpm@linux-foundation.org, clameter@sgi.com
Message-ID: <200802050735.m157ZUH3009868@imap1.linux-foundation.org>


From: Christoph Lameter <clameter@sgi.com>

I keep running upstream and mm kernels and the location of the slab
directory is different since upstream still uses /sys/slab.  This patch
makes slabinfo check /sys/slab if /sys/kernel/slab is not there.  Makes
slabinfo work on any kernel.

Signed-off-by: Christoph Lameter <clameter@sgi.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 Documentation/vm/slabinfo.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/Documentation/vm/slabinfo.c
+++ b/Documentation/vm/slabinfo.c
@@ -1123,7 +1123,7 @@ void read_slab_dir(void)
 	char *t;
 	int count;
 
-	if (chdir("/sys/kernel/slab"))
+	if (chdir("/sys/kernel/slab") && chdir("/sys/slab"))
 		fatal("SYSFS support for SLUB not active\n");
 
 	dir = opendir(".");
